File: src/test/java/io/tripled/social/client/application/DefaultFollowUserUseCaseTest.java
package io.tripled.social.client.application;

import io.tripled.social.client.domain.DateTimeProvider;
import io.tripled.social.client.infrastructure.TestSocialNetworkRepository;
import org.junit.jupiter.api.Test;

public class DefaultFollowUserUseCaseTest extends AbstractUseCaseTest {

  private DefaultFollowUserUseCase useCase;

  @Override
  protected void initializeUseCase(TestSocialNetworkRepository socialNetworkRepository, DateTimeProvider dateTimeProvider) {
    useCase = new DefaultFollowUserUseCase(socialNetworkRepository);
  }

  @Test
  void canFollow() {
    useCase.follow("Alice", "Bob");

    assertUserIsFollowing("Alice", "Bob");
  }

  @Test
  void canFollowTwoUsers() {
    useCase.follow("Alice", "Bob");
    useCase.follow("Alice", "Jane");

    assertUserIsFollowing("Alice", "Bob", "Jane");
  }
}

File: src/test/kotlin/io/tripled/social/client/application/DefaultFollowUserUseCaseTest.kt
package io.tripled.social.client.application

import io.tripled.social.client.domain.DateTimeProvider
import io.tripled.social.client.infrastructure.TestSocialNetworkRepository
import org.junit.jupiter.api.Test

class DefaultFollowUserUseCaseTest : AbstractUseCaseTest() {
    private var useCase: DefaultFollowUserUseCase? = null

    override fun initializeUseCase(
        socialNetworkRepository: TestSocialNetworkRepository,
        dateTimeProvider: DateTimeProvider
    ) {
        useCase = DefaultFollowUserUseCase(socialNetworkRepository)
    }

    @Test
    fun canFollow() {
        useCase!!.follow("Alice", "Bob")

        assertUserIsFollowing("Alice", "Bob")
    }

    @Test
    fun canFollowTwoUsers() {
        useCase!!.follow("Alice", "Bob")
        useCase!!.follow("Alice", "Jane")

        assertUserIsFollowing("Alice", "Bob", "Jane")
    }
}